lib/Parse/ParsePattern.cpp

```diff
@@ -1255,7 +1255,7 @@ ParserResult<Pattern> Parser::parsePatternTuple() {
   SmallVector<TuplePatternElt, 8> elts;
   ParserStatus ListStatus =
     parseList(tok::r_paren, LPLoc, RPLoc,
-              /*AllowSepAfterLast=*/false,
+              /*AllowSepAfterLast=*/true,
```

Comment:
I think it’s okay to allow it unconditionally, but we have to diagnose its presence when the feature is not enabled. Bonus points if it offers a fixit that removes the extraneous comma!